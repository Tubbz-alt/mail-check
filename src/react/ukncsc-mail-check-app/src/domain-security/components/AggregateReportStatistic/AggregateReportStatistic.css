.Statistic {
  border-left: 8px solid #082147;
  padding-left: 15px;
  margin-top: 35px;
}

.StatisticTitle {
  margin-bottom: 0px;
  font-weight: bold;
}

.StatisticValue {
  margin-top: 0px;
  font-weight: bold;
}

.StatisticPeriod {
  margin: 0;
  color: gray;
}
