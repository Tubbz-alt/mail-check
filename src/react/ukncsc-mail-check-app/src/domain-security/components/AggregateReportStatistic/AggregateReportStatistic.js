import React from 'react';
import { Header } from 'semantic-ui-react';
import numeral from 'numeral';

import './AggregateReportStatistic.css';

export default ({ title, value, period }) => (
  <div className="Statistic">
    <Header as="h3" style={{ margin: 0 }} className="StatisticTitle">
      {title}
    </Header>
    <Header className="StatisticValue" style={{ margin: 0 }} as="h1">
      {numeral(value).format('0.0a')}
    </Header>
    <p className="StatisticPeriod">{period}</p>
  </div>
);
