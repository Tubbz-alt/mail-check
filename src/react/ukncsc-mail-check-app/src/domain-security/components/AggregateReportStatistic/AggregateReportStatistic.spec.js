import React from 'react';
import { render } from 'react-testing-library';
import AggregateReportStatistic from './AggregateReportStatistic';

describe('AggregateReportStatistic', () => {
  let container;

  describe('when getting an aggregate report statistic', () => {
    beforeEach(() => {
      ({ container } = render(
        <AggregateReportStatistic
          title="stat"
          value="2000"
          period="last week"
        />
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());
  });
});
