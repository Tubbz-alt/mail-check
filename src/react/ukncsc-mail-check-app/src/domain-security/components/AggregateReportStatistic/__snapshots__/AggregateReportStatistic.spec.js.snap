// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AggregateReportStatistic when getting an aggregate report statistic it should match the snapshot 1`] = `
<div>
  <div
    class="Statistic"
  >
    <h3
      style="margin: 0px;"
      class="ui header StatisticTitle"
    >
      stat
    </h3>
    <h1
      style="margin: 0px;"
      class="ui header StatisticValue"
    >
      2.0k
    </h1>
    <p
      class="StatisticPeriod"
    >
      last week
    </p>
  </div>
</div>
`;
